src/tool/core/helpers/problem_code.rs

```diff
@@ -249,56 +254,66 @@ enum FunctionArgType {
 impl FunctionArgType {
     /// Applies any special changes needed to the value based on the type
     fn apply(&self, line: &str) -> anyhow::Result<String> {
+        debug!("Going to apply changes to argument input for {self:#?} to {line:?}");
         use FunctionArgType::*;
-        Ok(match self {
-            I32 => {
-                if let Err(e) = line.parse::<i32>() {
-                    warn!("In testing the test input \"{line}\" the parsing to i32 failed with error: {e}")
-                };
-                line.to_string()
-            }
-            I64 => {
-                if let Err(e) = line.parse::<i64>() {
-                    warn!("In testing the test input \"{line}\" the parsing to i64 failed with error: {e}")
-                };
-                line.to_string()
-            }
-            F64 => {
-                if let Err(e) = line.parse::<f64>() {
-                    warn!("In testing the test input \"{line}\" the parsing to f64 failed with error: {e}")
-                };
-                line.to_string()
-            }
-            VecI32 | VecBool | VecF64 => {
-                Self::does_pass_basic_vec_tests(line)?;
-                format!("vec!{line}")
-            }
-            VecString => {
-                Self::does_pass_basic_vec_tests(line)?;
-                let mut result = line.replace("\",", "\".into(),"); // Replace ones before end
-                result = result.replace("\"]", "\".into()]"); // Replace end
-                format!("vec!{result}")
-            }
-            VecVecI32 => {
-                Self::does_pass_basic_vec_tests(line)?;
-                line.replace('[', "vec![")
-            }
-            String_ | Bool => line.to_string(),
-            List => {
-                Self::does_pass_basic_vec_tests(line)?;
-                format!("ListHead::from(vec!{line}).into()")
-            }
-            Tree => {
-                Self::does_pass_basic_vec_tests(line)?;
-                format!("TreeRoot::from(\"{line}\").into()")
+        let result = match self {
+            I32 => match line.parse::<i32>() {
```

Comment:
derving  `#[derive(strum_macros::Display)]` allows to print the value as string.
we can merge these three lines maybe.

Reply:
I don't understand. It shows me 2 lines 
![image](https://github.com/rust-practice/cargo-leet/assets/43485962/687ede5a-7b87-430f-b76e-acd56f813d7c)

And I don't understand what you mean related to the lines I see.

Reply:
Going to move forward with this for now. Please open an issue with more info if this is still desired